packages/middleware-flexible-checksums/src/resolveFlexibleChecksumsConfig.ts

```diff
@@ -0,0 +1,33 @@
+import { Provider } from "@smithy/types";
+import { normalizeProvider } from "@smithy/util-middleware";
+
+import { DEFAULT_REQUEST_CHECKSUM_CALCULATION, DEFAULT_RESPONSE_CHECKSUM_VALIDATION } from "./constants";
+
+export interface FlexibleChecksumsInputConfig {
+  /**
+   * Determines when a checksum will be calculated for request payloads.
+   */
+  requestChecksumCalculation?: string | Provider<string>;
+
+  /**
+   * Determines when checksum validation will be performed on response payloads.
+   */
+  responseChecksumValidation?: string | Provider<string>;
+}
+
+export interface FlexibleChecksumsResolvedConfig {
+  requestChecksumCalculation: Provider<string>;
+  responseChecksumValidation: Provider<string>;
+}
+
+export const resolveFlexibleChecksumsConfig = <T>(
+  input: T & FlexibleChecksumsInputConfig
+): T & FlexibleChecksumsResolvedConfig => ({
+  ...input,
```

Comment:
Not something to address for this PR, but overall I think we should mutate directly instead of shallow-copy-and-write Object.assign.  

This is a config resolver and not in a hot path so it's fine here.